packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -44,7 +48,83 @@ export function buildRootObject() {
 
     await getPromiseForIssuerRecords(issuers);
 
-    const allSeatStagings = new Set();
+    const allSeatStagings = new WeakSet();
+
+    /** @type MakeZCFMint */
+    const makeZCFMint = async (keyword, mathHelperName = 'nat') => {
+      const zoeMintP = E(zoeInstanceAdmin).makeZoeMint(keyword, mathHelperName);
+      const mintyIssuerRecord = await E(zoeMintP).getIssuerRecord();
+      // AWAIT
+      const { brand: mintyBrand } = mintyIssuerRecord;
+      const mintyAmountMath = makeAmountMath(mintyBrand, mathHelperName);
+      // TODO some zcf-side registration of the new issuer
+
+      /** @type ZCFMint */
+      const zcfMint = harden({
+        getIssuerRecord: () => {
+          return mintyIssuerRecord;
+        },
+        mintGains: (gains, zcfSeat = undefined) => {
+          assert(
+            zcfSeat !== undefined,
+            details`On demand seat creation not yet implemented`,
+          );
+          let totalToMint = mintyAmountMath.getEmpty();
+          const oldAllocation = zcfSeat.getCurrentAllocation();
+          const updates = objectMap(gains, ([seatKeyword, amountToAdd]) => {
+            totalToMint = mintyAmountMath.add(totalToMint, amountToAdd);
+            const oldAmount = oldAllocation[seatKeyword];
+            // oldAmount being absent is equivalent to empty.
+            const newAmount = oldAmount
+              ? mintyAmountMath.add(oldAmount, amountToAdd)
+              : amountToAdd;
+            return [seatKeyword, newAmount];
+          });
+          const newAllocation = harden({
+            ...oldAllocation,
+            ...updates,
+          });
+          const zcfSeatAdmin = seatToZCFSeatAdmin.get(zcfSeat);
+          // verifies offer safety
+          const seatStaging = zcfSeat.stage(newAllocation);
+          // No effects above. Commit point. Must mint and commit atomically.
+          // (Not really. If we minted only, no one would ever get those
```

Comment:
I didn't understand this comment. It seems to argue with itself, and I wasn't sure which side won :)

Reply:
Clarified in the next PR